// flow-parser/flow_parser.js
// Generated by js_of_ocaml
//# buildInfo:effects=false, kind=exe, use-js-string=true, version=5.1.1+git-v0.24.0-13395-g0a538f2
(function(a){typeof
globalThis!=="object"&&(this?b():(a.defineProperty(a.prototype,"_T_",{configurable:true,get:b}),_T_));function
b(){var
b=this||self;b.globalThis=b;delete
a.prototype._T_}}(Object));(function(ak){"use strict";var
kJ="symbol",LP=271,nB=126548,yh=71127,fU="member_property",yg=65007,DL=66517,fT="jsx_attribute_value_expression",DK=119980,dH="function_declaration",bJ="<2>",A$=68466,KY="%=",DJ="??",vo=70080,vp="&",b4="identifier",p6=72163,yf=71723,A_="properties",DI=183969,vn=68223,fS="function_return_annotation",DH=124903,A9=70106,DG=241,KX="(",I2=213,ye=120074,vm=70708,A8=71679,fR="logical",aS="camlinternalFormat.ml",eN="type_guard_annotation",A7=92975,KW=">>>",vl="RestElement",yd=67897,JX=179,vk="start",A6=113775,o9=126521,JW="%i",KV="#",DF=43702,p5=126,yc=110947,g3="function_identifier",p4=119893,yb=70366,DC=65547,DD=43743,DE=-43,dG=8238,I1=",",fQ=8286,fP="keyof_type",vj=66717,o8=12336,LO=201,vi=71338,KU=270,nA=11565,DB=69289,vh=55291,vg=73030,ya=70479,Dz=69572,DA=11623,g2="tuple_spread_element",cl=8239,vf=64310,jc="@]",x$=42993,o7=11558,p3="Map.bal",kI="public",LN=-32,jb="Literal",dF="jsx_member_expression_identifier",eM="for_in_assignment_pattern",A5=71450,o6=126557,b9=103,x_=12292,ve=110579,or=120597,A3=13311,A4=12348,fO="export_default_declaration_decl",g1="tuple_type",vd=113663,x9=170,x8=67413,JV="Assert_failure",KT="comments",k9="%S",cp=127343600,A2=12341,x7=67646,LM="ENOTEMPTY",x6=72160,Dy=70187,I0=222,A1=12343,KS=2147483647,vb=126624,vc=43442,Dx=70312,dE="interface_type",iC="new",x5=66256,oq=68296,A0=124908,oo=126579,op=70107,AZ=249,va=71167,eL="union_type",c=248,u$=126546,dD="enum_bigint_member",KR=133,u9=67871,u_=66955,KQ=1027,g0="class_declaration",eK="optional_call",JU="a string",p2=11703,IZ="<<",o5=126564,dC="jsx_element",eJ="object_property_type",u8=94207,dB="enum_declaration",Dw=68023,x3=67669,x4=8318,JT="prefix",iv="this",x2=126578,eI="if_consequent_statement",JS=-696510241,AY=66963,iq="default",p1=72967,b8=101,JR="buffer.ml",x1=74649,co=0xffff,AX=43709,on=175,LL="===",kH=117,eH="jsx_identifier",IY="EnumDefaultedMember",nz=70006,AW=70161,x0=126633,Dv=66965,eG="member_property_expression",Du=101589,xZ=64274,jw="function",xX=66303,xY=42954,AV=126529,Dt=72191,dA="new_",u7=64433,ny=126559,xW=72144,JQ="==",hX=-744106340,Ds=43359,u6=171,KP="Printexc.handle_uncaught_exception",u5=66735,AU=126534,xV=74879,u4=42785,om=120629,k3="0o",KO="End_of_file",u3=66175,JP="&=",AT="nan",p0=126503,xU=43470,fN="import_namespace_specifier",xT=77711,ol=70302,dz="component_param",ok="@])",pZ=126515,k2=118,bG="continue",AS=43798,o4=";@ ",Dr=74751,u2="src/parser/statement_parser.ml",Dq="rmdir",xS=94177,gZ="for_in_statement",LK=12520,IX="TypeParameterInstantiation",w="",JO="**=",Dp=120126,IW=197,AR=67829,nx="_bigarr02",gY="export_named_declaration_specifier",pY=": No such file or directory",pX=64319,u1=69926,dy="pattern_object_p",IV="TypeAnnotation",fM="array_type",kG="@[%s =@ ",xR=72847,dx="export_default_declaration",oj=126590,AP=42774,AQ=": Not a directory",ji="let",gX=12288,bo="argument",bq="/",o3="an identifier",hq="typeof",oi=68116,xQ=182,fL="declare_export_declaration_decl",u0=67589,uZ=66771,kU="class",IU="tokens",oh=70281,kT=255,AO=43638,iu="key",Do=69955,IT=">>",dw="function_expression_or_method",xP=43587,cn="block",xO=100351,o2="mixed",AN=66503,xN="ENOTDIR",JN=-15,AM=65135,fK="string_literal",bv="@ ",xM=43334,eF="if_alternate_statement",AL=70448,nw=8485,eE="type_args",uY=69864,dv="if_statement",LJ="+=",gW="typeof_identifier",nv="with",pW=65595,xL=64286,AK=71086,gV="true",Dn=69423,gU="catch_clause",uX="e",jv="asserts",IS=">>=",JM=131,uW=43388,xK=43887,aF=-48,Dm=120779,LI=190,uV=194,AI=71351,AJ=65629,fJ="call",LH=-42,nu=126553,AH=43695,AG=177,uU=42124,xJ=12703,AE=12442,AF=11718,o1=70449,og=126547,uT=67462,hm="left",eD="infer_type",Dl=11742,xI=65597,of="Unix.Unix_error",AD=122623,Dj=124911,Dk=72959,xH="opaque",gT="object_internal_slot_property_type",JL="Enum `",fI=65279,uR=71983,uS=12329,at=110,du="spread_property",JK="importKind",k1=" =",gS="remote_identifier",eC="labeled_statement",dt="jsx_fragment",Di=120770,ds="function_param",a_=112,Dh="exportKind",eB="binary",hk="`.",Dg=42511,IR="<=",eA="jsx_spread_attribute",bd="import",fH="typeof_member_identifier",Df=69414,xG=19967,nt=11687,xF=93823,LG=209,De=71903,uQ="of",AC=72000,Dd="typeArguments",ey="type_identifier",ez="pattern_array_element_pattern",o0=69744,ja=192,dr="class_element",fF="export_source",fG="component_param_pattern",AB=42508,Dc=125124,IQ="Unexpected token `",dq="for_in_left_declaration",gR="object_call_property_type",xE="abstract",uP=8584,AA=68786,uO=71999,oZ=123214,xC=123565,xD=186,gQ="class_implements_interface",oY=126536,Db=69749,IP="Invalid legacy octal ",Ay=71295,Az=66927,ex="pattern_expression",Ax=11679,Da=-61,pV=65141,xB=11694,KN="\r\n",gP="update_expression",IO="minus",bI="debugger",C$=71352,uN=65470,jl="number",C_=123627,pU=64322,oe=43471,fE="for_of_assignment_pattern",od=126589,xA=43784,KM="Internal Error: Found object private prop",C9=183983,b="id",Aw=123190,oc="finally",oX=120070,C8=72095,dp="syntax",xz=110591,fD="false",LF=-10,ns="AssignmentPattern",dn="typeof_expression",Av=43764,IN="FunctionTypeParam",dm="function_body_any",At=126627,Au=71998,C7=126543,dl="call_type_arg",xy=64316,oW=64285,C6=8454,KL=137,IM="**",fC="object_type_property_setter",C5=68607,hO=108,C4="out",xx=68799,fA=65278,fB="jsx_member_expression",As=92728,hW="null",Ar=66431,C3=72249,cs=0x80,nr=119994,xw=66207,C2=43583,oU="else",oV=94179,pT=11735,C1=64911,gO="jsx_attribute_name_namespaced",JJ="!",C0=42539,uM=72250,Aq=71215,Ap=69746,uL=65487,gN="pattern_object_property_key",IL=", ",An=8505,Ao="=",Am=64111,CZ=8507,pS=120134,ob="while",CX=120596,CY=43002,k8="protected",uK=68479,xv=43395,Al=68252,LE="v",KK=152,uJ=70278,Ak=70853,oT=120145,uI=69297,Aj=73112,pR=8488,uG=68351,uH=42655,gM="for_of_left_declaration",xt=44031,xu="Failure",Ai=92159,fz="object_key_identifier",IK=195,iB="bigint",dk="import_default_specifier",kS=256,fy="member",LD="!==",dj="component_identifier",CW=73008,CV=72283,pQ=126500,pP=120127,ev="jsx_attribute_name",ew="for_statement_init",Ah=67711,fx="private_name",oa="case",nq=8489,eu="import_specifier",CU=64279,Ag=94098,xs=119974,uF=72969,IJ=193,JI="!=",np=126520,uE=71944,uD=42191,di="generic_qualified_identifier_type",it="implements",Af=194559,KJ="%",kF="hasUnknownMembers",CT=71039,LC=211,Ae=83526,fv="init",fw="jsx_attribute_value",Ad=70271,ju=240,et="function_type_return_annotation",Ac=70018,gL="readonly_type",xr=512,CS=68095,xq=120003,n$=126563,n_=71236,uC=69375,Ab=68850,xp=70105,uB=43866,oS=888960333,Aa=43013,gK="assignment_pattern",xo="specifiers",LB=710,hS="as",z$=120570,CR=11507,II=260,KI=204,fu="jsx_element_name_identifier",gJ="class_expression",IH=257,CQ=44002,z_=82943,i$="src/parser/type_parser.ml",jt="test",z9=64217,pO="package",KG="collect_comments",KH="Pervasives.do_at_exit",CO=125183,CP=42606,gI="tuple_element",ft="enum_boolean_member",uA=65312,no=119981,CN=65495,pN=120085,LA=-80,Lz=138,n9=126555,uz=65276,an=128,IG="{ ",es="for_statement",fs="class_method",pM="if",fr="generic_type",X=113,z8=43071,xn=72001,CM=71131,z7=70002,xm=42888,nn=8469,kE="instanceof",xl=11502,oR=94178,uy=64321,xk=64913,JH="Division_by_zero",z6=92879,ux=71945,IF=185,z5=66938,pL=65535,z4=113800,Ly=": file descriptor already closed",pK=223,Lx="*=",uw=68899,er="switch_case",fq="pattern_array_element",gH="enum_string_member",Lw="visit_trailing_comment",oQ="export",nm=120122,pJ=43823,xi=43792,xj=42527,CL=70726,dh="enum_defaulted_member",z3=68497,pI=72349,gG="program",eq="member_type_identifier",z2="object",gF="for_of_statement_lhs",z1=113791,uv=67391,dg="jsx_spread_child",n7=126554,n8=8526,n6=43880,oP=69415,CK=43822,df="pattern_identifier",z0=93052,js="readonly",bT="name",xh=68119,zZ=71494,CJ=120121,nl=8486,IE=2047,fp="enum_symbol_body",JG=142,KF="PropertyDefinition",xg=177976,gE="declare_class",xf=65489,zX=72367,zY=70440,de="import_named_specifier",KE="Popping lex mode from empty stack",uu=68111,xe=66463,ID="*-/",xd=43187,nk=8487,zW=11567,ut=67861,KD=159,xc="\n",xb=66383,gD="declare_interface",us=-24976191,KC=238,JE=-24,JF="@ }@]",zV=43645,Lv=176,ur=119976,nj=69959,zU=126519,JD=";",KB="trailingComments",pH=65548,ep="number_literal",i_=449540197,xa=43704,n5=126584,uq=8467,Lu="||",ni=11695,zT="exported",w$=120712,hp="void",JC="mixins",w_=92783,CI=215,JB="%ni",jh=">",fo="jsx_child",CH=8516,dd="optional_indexed_access_type",eo="typeof_type",en="spread_element",zS=42963,Lt="@[",gC="component_params",zR=43042,nh="\x01\x02",fn="function_",pG="for",n4=65575,hT="params",zQ=168,Ls="win32",dc=8202,JA="@",n3="^",Lr=164,b6="optional",up=65574,kD="boolean",KA=139,uo=12548,oO=120539,Lq="Not_found",ng=246,gB="expression_statement",CG="EBADF",un=66815,um=55203,db="function_param_type",zP=73064,pF=70279,w9=110580,Lp=233,n2="<",Lo="visit_leading_comment",w8=66855,zO=66966,ul=66499,w7=111355,zN=68680,Ln=206,Jz="--",w6=65497,pE=11711,fm="function_param_pattern",is="constructor",jk=5760,Kz="infinity",oN=43642,zM="fs",CF=92991,pD=126544,w5=101640,oM=72162,CE=67583,oL=8468,bx="typeParameters",Ky="elements",w4=71423,Lm="Sys_blocked_io",gA="interface_declaration",em="variable_declaration",da="function_rest_param",el="type",zL="Invalid number ",zK=" : flags Open_rdonly and Open_wronly are not compatible",uk=69404,c$="jsx_element_name_member_expression",oK="keyof",n1="never",ek="with_",ck=0x8000,Kx="|=",oJ=70404,pC=70441,zJ=42969,kC="declare",zI=73061,fl="object_type",gz="object_property_value_type",w3=69687,Kw="Invalid binary/octal ",uj=230,zH=64324,Kv="range",IC="infer",CD=120744,gy="array_element",ui=70730,zG=43641,Ku=166,CC=70461,uh=69890,zF=69487,uf=74862,ug=68149,nf=73065,Jy="%a",w2=72348,IB=172,c_="jsx_expression",CB=65663,ue=126495,zE=245,CA=124907,gx="member_property_identifier",IA=226,zD=43615,fk="comment",ne=119965,ej="catch_clause_pattern",fj="object_type_property_getter",Iz=136,w1=43019,Cz=67455,pB=126628,Jx=331416730,Ll="the start of a statement",ud=122654,zC="shorthand",Cy=43595,Cx=11710,gw="typeof_qualified_identifier",w0=72750,Jw="elementType",aY="typeAnnotation",zB=124895,Kt=162,pA=11559,wZ=67382,Ks="??=",uc=72329,wY="target",Kr=180,Jv=189,Kq=8206,ub=43513,wX=173823,ua=126467,c9="type_guard",wV=43700,wW=12783,oI=8305,fi="type_annotation",Cw=141,bO="break",nd=42999,t_=65019,t$=160,Cv=70460,c8="expression_or_spread",t9=")",fh="class_private_field",Cu=55215,t8=65338,zA=40981,k0="members",ei="import_declaration",wT=69634,n0=94031,wU="ENOENT",Ct=8457,Iy="satisfies",fg="generic_identifier_type",gv="function_this_param",wS=66993,eh="type_",zz=67423,t7=11557,wR=12799,nc=239,wQ=93026,t6=66377,zy=123180,Kp=221,Lk=-594953737,t5=67967,zx=43586,aX=105,Ix="src/parser/flow_lexer.ml",wP=66559,gu="class_property_value",zw=67637,Ko="closedir",t4=43010,zv=8521,oH=69956,wO=42959,Kn=212,zu=92735,nb="}",kR="method",wN=11498,nZ=247,bw="empty",gt=0xffffff,t3=161,zt=42887,ff="object_key_literal",iA=116,gs="type_identifier_reference",nY=126634,Cs=68029,Km="regexp",t2=70414,kQ=121,fe="template_literal_element",t1=8449,Cr=126562,c7=12287,jg=-45,nX=64297,nW=126523,zs=43301,cr=111,Cq=126498,t0=43776,Iw="EEXIST",Cp=119892,Co=43807,Kl=4096,jr=252,ho=0xff,Cn=68295,c5="variable_declarator_pattern",c6="do_while",nV="catch",tZ=66962,Cm=120654,gr=125,fd="label_identifier",Cl=11263,tY=8525,eg="assignment",Ck=191456,tX=43273,Lj="%u",tW=65381,wL=110927,wM=65479,tV=120538,ip="await",zq=71487,zr="jsError",zp=110588,Ci=120084,Cj=42890,jq=224,gq="object_key",zo=43696,tU=73647,wJ=43761,wK=12295,tT=64967,na=11647,wI=191,bU=123,Ch="generator",wH=123583,ef="for_of_statement",fc="enum_bigint_body",tR=110959,tS=92995,tQ=120686,Cg="b",wG=119969,oG=126522,nU=64318,zn=71839,pz=126602,tP=65908,kB=65536,Ju=231,Js=-602162310,Jt="comment_bounds",ee="-",Kk=-55,fb="pattern_object_property",tO=43493,tN=69505,wF=8471,wE=187,nT=120745,fa="enum_member_identifier",Ce=71959,Cf=66863,tM=65594,zm=253,k7='"',m$=70286,ed="jsx_attribute_value_literal",zl=68447,Jr="the",zk="index out of bounds",e$="declare_export_declaration",ec="jsx_attribute",e_="class_extends",ad=122,aE=106,gp="binding_pattern",eb="ts_type_cast",zj=113807,tK=93951,kA=119,tL="types",zi=8335,go="statement_fork_point",jf="_",wD=65500,e9="function_type",wC=68220,ea="statement_list",i9=-835925911,zh=123535,py=43815,Iv=199,oF=120571,wB=67514,Jq="Property",px=72713,Li="Unexpected ",Kj=169,Lh=", characters ",m_=43867,c4="pattern_literal",Cd=42537,d$="component_declaration",Ki=" : is a directory",Lg=5761,Z=127,aR=-36,kO=912068366,kP="delete",e8=114,Cc=120076,wA=65370,Cb=65481,am="value",tJ=68405,kN="operator",k6="const",hl=109,pw="any",tI=69958,wz=70831,zg=73111,Ca=72767,zf="Identifier",e7="jsx_opening_attribute",c3="conditional_type",wy="loc",wx=67071,nS=120004,tH=43492,ze=70005,Lf=188,pv=72272,tG=11389,ww=251,zd=73055,oE=70280,nR=1114111,tF=66421,Iu="Stack_overflow",wv=70301,zc=19903,iz="0x",wu=69967,zb=12447,m9=66512,Jp="Fatal error: exception %s\n",kZ=1000,B$=69295,pu=120093,Jo=">=",oD=64325,e6="class_identifier",tD=119967,tE=68415,Jn="end",d_="enum_boolean_body",e5="member_private_name",d9="super_expression",za=71955,B_=126514,m8=67593,wt=66939,B9=12591,pt=126538,tC=110590,B8=72703,tB=72105,nQ=65598,B7=73727,ps=126504,m7=126551,ws=70143,kM="from",gn="class_property",d8="enum_number_body",tA=42559,wq=93759,wr=66994,cq="right",Le=225,tz=67702,wo=65473,wp=43697,nP=70855,ty=119993,tx=72103,wn=178205,d7="call_type_args",y$=66511,gm="export_batch_specifier",cm=782176664,iy="get",kz="local",wm=228,gl="object_mapped_type_property",c2="class_decorator",Jm=220,c1="enum_body",Ld="<<=",wl=71956,y_=69839,jj="super",y9=173791,B6=71942,aK="expression",tw=72440,je=254,wk=70412,d6="try_catch",Kh=0x7ff0,d5="declare_module_exports",tv=12320,Kg=134,B5=94175,kY="enum",It=196,c0="import_source",y8=43814,wj=120069,d4="while_",oC=126537,tu=43262,e4="function_rest_param_type",y7=66378,pr=119996,wi=73097,y5=70783,y6=43503,wh=131071,B4=11492,tt=92766,Kf=173,B2=113770,B3=73029,ts=66978,gk="tagged_template",d2="jsx_element_name",d3="for_init_declaration",wg=123213,cZ="object_indexer_property_type",d1="object_spread_property_type",oA=72970,oB=70854,wf=110930,kL="var",gj="pattern_object_property_literal_key",Is=217,B1=119972,we=69622,y3=63743,y4=42237,tr=870530776,ky="returnType",Ir=0xdc00,jp="computed",tq=42735,e3="arg_list",wd=67461,e2="export_named_declaration",tp=72817,tn=73439,to=43782,tm=66775,y2=70655,oz="bool",B0=65140,tl=75075,wb=126651,wc=71947,tk=42961,tj=12735,wa=78894,v$=64262,Lc=237,bp="interface",Lb="Match_failure",v_=42962,y1=69748,Iq="leadingComments",cY="this_expression",kX=461894857,pq=12592,Ip=8204,BZ=119807,v9=66348,cX="declare_variable",y0=8348,e1="optional_member",d0=120,e0="arrow_function",ti=72768,BY=70851,eZ="array",th=43249,nO=126468,tg=177983,yZ="compare: functional value",tf=126550,yY=64847,gi="binding_type_identifier",v8=120132,gh="function_params",te=93071,kK=1024,BX=42783,Ke=1039100673,Jl="@{",v7=12352,BW=42653,v6=120628,gg="declare_function",eY="for_in_statement_lhs",BV=72271,td=69807,v5=67826,eX="syntax_opt",tc=243,BU=94032,In=265,Io="Undefined_recursive_module",Im=-1053382366,tb=72242,cW="variance_opt",yW=101631,yX="arguments",ta=72161,v4=8511,pp="unknown",s$=43560,Ik="the end of an expression statement (`;`)",Il=1026,yV=12543,BT=11670,yU=69247,po=11631,nN="line",v3=72202,dZ="literal",cV="pattern_object_rest_property",b7=" ",yT=43487,hV=115,v2=-673950933,dY="intersection_type",s_=120144,kx="is",v0=178207,v1=100343,Kd="||=",La="f",pn=8455,cj=102,yS=70418,BS=8543,K$="Internal Error: Found private field in object props",oy=126540,pm=119995,ci=8287,cU="indexed_access_type",dX="export_named_specifier",Kc=266,Jk=0xe0,yR=124926,s8=-103,s9=167,ox=65344,nM=126530,BR=113788,s6=67505,s7="property",yQ=43014,bN="return",io=-85,s5=126601,Ij=214,Jj="children",eW="type_alias",m6=43259,vX=126583,vY=71958,vZ=65613,yP=67431,ow=126535,vW=69599,dV="type_params",dW="object_key_computed",vV=124910,be="variance",m5=11727,BQ=66954,s4=126463,dU="catch_body",vU=69445,dT="type_param",BP=124902,m4=120687,Kb="collect_comments_opt",BO=0xf,yO=120485,BN=70416,BM=125259,gf="jsx_namespaced_name",yN=43712,BL=72712,Ka="~",pl=12448,ge="jsx_member_expression_object",m3=126499,nL=-97,dS="pattern_object_property_identifier_key",Ji=219,eV="component_body",gd="opaque_type",cT=".",vT=43009,yM="consequent",Jh="SpreadElement",aB="body",J$=178,K_=202,gc="jsx_opening_element",cS="declare_module",m2=67638,BK=8477,dR="object_type_property",vS=110882,dQ="function_body",s3=94111,Jg="module",Ii="alternate",yL=67839,cR=8191,s2=43881,Jf=": closedir failed",hR="kind",cQ="tuple_labeled_element",ix=-46,BJ=67640,gb="declare_type_alias",ov=70750,s1=77808,kw="column",eU="jsx_closing_element",s0=66977,vR="\x01\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\x01",sZ=65786,ga="function_expression",ch=104,m1=11719,BI=11505,yK="mkdir",sY=70319,Je="Invalid_argument",BH=43738,vQ=113817,eT="bigint_literal",sX=70084,yJ=126566,m0="do",pk=42622,f$="computed_key",eS="pattern_object_property_computed_key",Jd="fd ",mZ=126571,vP=126619,BG=140,hU="prototype",sW=208,sV=67004,Ih=130,nK=242,Jc=">>>=",yI=68863,sU=11726,b5="raw",yH=64466,hj=107,sT=67679,cP="enum_string_body",yG=244,K9="*",sS=66335,yF=126570,J_=229,yE=" : file already exists",yD=184,BF=67807,vO=70753,f_="boolean_literal",BE=65437,sR=70451,vN=67002,jo=124,f9="conditional",nJ=43260,J9="Sys_error",BD=123135,kv="meta",BC=64109,f8="pattern_array_rest_element",BB=43255,pj=67644,eR="pattern_object_rest_property_pattern",f7="sequence",vM=65855,Bz=110951,BA=67643,f6="predicate_expression",bM="static",K8=263,yC=120512,yB="declaration",pi=64317,yA=68437,ph=126558,f5="meta_property",By=11564,f4="declare_enum",cg="pattern",yz=216,sQ=68191,mX="undefined",mY=8319,nI=120133,Jb=132,sP=42239,Ig=-99,vL=124927,yy=120092,Bx=43137,dP="component_rest_param",J8="expected *",Bw=125251,K7="%li",sN=55242,sO=12294,cO="enum_number_member",hQ="in",Ja="\\\\",ir=":",sM=68115,If="Cygwin",vK=77823,yx=65615,pg=70162,Ie="/static/",yw=11519,vJ=72966,yu=12686,J7=165,yv=183,jn=129,mW=72192,Bv=42964,mV="try",nH=120655,vI=11702,K6="expressions",Bu=2048,dO="class_body",Bt=55238,J6=0xf0,sL=66915,Bs=43311,sK=43018,pf=235,sJ=73648,Id="([^/]+)",yt=125258,Br=64829,vH=68735,I_="++",I$="{",J5=163,I9="qualification",K5=0xdfff,Ic=931,ys=71235,Ib=8472,sI=71934,Ia=205,yr=218,J4="callee",Bo=43711,Bp=64284,Bq=43754,yq=43790,J3="%Li",f3="pattern_array_rest_element_pattern",nG="decorators",yp=8304,f2="statement",nF=73062,dN="jsx_children",yo=70492,Bn=64255,sH=11630,vG=1255,ou=67592,mU=43519,nE=64311,pe=12539,vF="proto",nD=120513,yn=68031,iw="source",k5="a",sG=93047,ym=92927,Bl=126588,Bm=73458,sF=67742,Bk=43714,vE=236,Bj=-253313196,k4="label",K4="@[<2>{ ",pd=126539,mT=126552,vD=120487,K2=268,K3="Out_of_memory",Bi=605857695,yl=94026,pc=126496,hP="async",J2=203,pb=126560,H$=156,vC=68287,eQ="unary_expression",sD=-26065557,sE=110587,mS=120771,vB=69762,vA=126502,i8="set",eP="object_",f1="template_literal",sC=43258,f0="nullable_type",hn="int_of_string",H_="^=",bL="predicate",jd="string",ot=8450,I8="camlinternalMod.ml",yk=70285,hi="+",Bh=110575,H9=198,fZ="extends",I7=-692038429,mR=67827,K1=210,vz=227,os="explicitType",mQ=70452,vy=70497,jm=0x3f,kW="private",sB=64296,yj=67591,Bg=92909,vx="Fatal error: exception ",Bf=120137,mP=68120,dM="pattern_array_e",Bd=119964,Be=92862,sA=66461,H8="&&=",J1=174,a9=8231,J0="/=",vw=66811,mO=70108,vv=67504,Bc=11686,sy=67001,sz=" : flags Open_text and Open_binary are not compatible",Bb=43741,sx=66204,aZ=8233,cN="type_annotation_hint",sw=123197,cM="object_property",mN=70480,I6="&&",fY="type_cast",kV="%d",pa=8484,K0=207,vt=70066,vu=68324,nC=120713,I5=135,KZ="'",o$=126556,bn="0",bc="yield",mM=126591,eO=100,Ba=69551,fX="jsx_element_name_namespaced",H7=232,dL="function_this_param_type",fW="pattern_object_property_pattern",bH="throw",bK="switch",JZ=0x800,o_=119970,dK="toplevel_statement_list",im=250,vs=12438,dJ="class_implements",fV="variable_declarator",sv=43713,mL=68096,vr=70457,su=12538,vq=11734,I4="-=",H6=234,cL="component_param_name",st=43123,cK="class_",JY="|",I3=200,sr=43518,ss=8483,dI="jsx_attribute_name_identifier",yi=181;function
atn(a,b,c,d,e){if(d<=b)for(var
f=1;f<=e;f++)c[d+f]=a[b+f];else
for(var
f=e;f>=1;f--)c[d+f]=a[b+f];return 0}function
ato(a){var
c=[0];while(a!==0){var
d=a[1];for(var
b=1;b<d.length;b++)c.push(d[b]);a=a[2]}return c}function
atp(a,b,c){var
d=new
Array(c+1);d[0]=0;for(var
e=1,f=b+1;e<=c;e++,f++)d[e]=a[f];return d}function
qi(a,b,c){var
d=String.fromCharCode;if(b==0&&c<=Kl&&c==a.length)return d.apply(null,a);var
e=w;for(;0<c;b+=kK,c-=kK)e+=d.apply(null,a.slice(b,b+Math.min(c,kK)));return e}function
p9(a){var
c=new
Uint8Array(a.l),e=a.c,d=e.length,b=0;for(;b<d;b++)c[b]=e.charCodeAt(b);for(d=a.l;b<d;b++)c[b]=0;a.c=c;a.t=4;return c}function
hs(a,b,c,d,e){if(e==0)return 0;if(d==0&&(e>=c.l||c.t==2&&e>=c.c.length)){c.c=a.t==4?qi(a.c,b,e):b==0&&a.c.length==e?a.c:a.c.substr(b,e);c.t=c.c.length==c.l?0:2}else if(c.t==2&&d==c.c.length){c.c+=a.t==4?qi(a.c,b,e):b==0&&a.c.length==e?a.c:a.c.substr(b,e);c.t=c.c.length==c.l?0:2}else{if(c.t!=4)p9(c);var
g=a.c,h=c.c;if(a.t==4)if(d<=b)for(var
f=0;f<e;f++)h[d+f]=g[b+f];else
for(var
f=e-1;f>=0;f--)h[d+f]=g[b+f];else{var
i=Math.min(e,g.length-b);for(var
f=0;f<i;f++)h[d+f]=g.charCodeAt(b+f);for(;f<e;f++)h[d+f]=0}}return 0}function
jA(a,b){if(a==0)return w;if(b.repeat)return b.repeat(a);var
d=w,c=0;for(;;){if(a&1)d+=b;a>>=1;if(a==0)return d;b+=b;c++;if(c==9)b.slice(0,1)}}function
p_(a){if(a.t==2)a.c+=jA(a.l-a.c.length,"\0");else
a.c=qi(a.c,0,a.c.length);a.t=0}function
D8(a){if(a.length<24){for(var
b=0;b<a.length;b++)if(a.charCodeAt(b)>Z)return false;return true}else
return!/[^\x00-\x7f]/.test(a)}function
Mh(a){for(var
j=w,d=w,g,f,h,b,c=0,i=a.length;c<i;c++){f=a.charCodeAt(c);if(f<cs){for(var
e=c+1;e<i&&(f=a.charCodeAt(e))<cs;e++);if(e-c>xr){d.substr(0,1);j+=d;d=w;j+=a.slice(c,e)}else
d+=a.slice(c,e);if(e==i)break;c=e}b=1;if(++c<i&&((h=a.charCodeAt(c))&-64)==an){g=h+(f<<6);if(f<Jk){b=g-0x3080;if(b<cs)b=1}else{b=2;if(++c<i&&((h=a.charCodeAt(c))&-64)==an){g=h+(g<<6);if(f<J6){b=g-0xe2080;if(b<JZ||b>=0xd7ff&&b<0xe000)b=2}else{b=3;if(++c<i&&((h=a.charCodeAt(c))&-64)==an&&f<0xf5){b=h-0x3c82080+(g<<6);if(b<0x10000||b>0x10ffff)b=3}}}}}if(b<4){c-=b;d+="\ufffd"}else if(b>co)d+=String.fromCharCode(0xd7c0+(b>>10),Ir+(b&0x3FF));else
d+=String.fromCharCode(b);if(d.length>kK){d.substr(0,1);j+=d;d=w}}return j+d}function
hr(a,b,c){this.t=a;this.c=b;this.l=c}hr.prototype.toString=function(){switch(this.t){case
9:return this.c;default:p_(this);case
0:if(D8(this.c)){this.t=9;return this.c}this.t=8;case
8:return this.c}};hr.prototype.toUtf16=function(){var
a=this.toString();if(this.t==9)return a;return Mh(a)};hr.prototype.slice=function(){var
a=this.t==4?this.c.slice():this.c;return new
hr(this.t,a,this.l)};function
LX(a){return new
hr(0,a,a.length)}function
hw(a){return a}function
g4(a){return LX(hw(a))}function
iE(a,b,c,d,e){hs(g4(a),b,c,d,e);return 0}function
Mi(a){var
b=ak.process;if(b&&b.env&&b.env[a]!=undefined)return b.env[a];if(ak.jsoo_static_env&&ak.jsoo_static_env[a])return ak.jsoo_static_env[a]}var
D5=0;(function(){var
c=Mi("OCAMLRUNPARAM");if(c!==undefined){var
b=c.split(I1);for(var
a=0;a<b.length;a++)if(b[a]==Cg){D5=1;break}else if(b[a].startsWith("b="))D5=+b[a].slice(2);else
continue}}());var
bg=[0];function
atx(a,b){if(!a.js_error||b||a[0]==c)a.js_error=new
ak.Error("Js exception containing backtrace");return a}function
s(a,b){return D5?atx(a,b):a}function
at3(a,b){throw s([0,a,b])}function
cw(a){return a}function
D4(a,b){at3(a,cw(b))}function
aM(a){D4(bg.Invalid_argument,a)}function
att(){aM(zk)}function
ac(a,b,c){c&=ho;if(a.t!=4){if(b==a.c.length){a.c+=String.fromCharCode(c);if(b+1==a.l)a.t=0;return 0}p9(a)}a.c[b]=c;return 0}function
jy(a,b,c){if(b>>>0>=a.l)att();return ac(a,b,c)}function
by(a,b){switch(a.t&6){default:if(b>=a.c.length)return 0;case
0:return a.c.charCodeAt(b);case
4:return a.c[b]}}function
ht(d,c){var
f=d.l>=0?d.l:d.l=d.length,e=c.length,b=f-e;if(b==0)return d.apply(null,c);else if(b<0){var
a=d.apply(null,c.slice(0,f));if(typeof
a!=="function")return a;return ht(a,c.slice(f))}else{switch(b){case
1:{var
a=function(a){var
f=new
Array(e+1);for(var
b=0;b<e;b++)f[b]=c[b];f[e]=a;return d.apply(null,f)};break}case
2:{var
a=function(a,b){var
g=new
Array(e+2);for(var
f=0;f<e;f++)g[f]=c[f];g[e]=a;g[e+1]=b;return d.apply(null,g)};break}default:var
a=function(){var
e=arguments.length==0?1:arguments.length,b=new
Array(c.length+e);for(var
a=0;a<c.length;a++)b[a]=c[a];for(var
a=0;a<arguments.length;a++)b[c.length+a]=arguments[a];return ht(d,b)}}a.l=b;return a}}function
k$(){aM(zk)}function
ay(a,b){if(b>>>0>=a.length-1)k$();return a}function
atv(a){if(isFinite(a)){if(Math.abs(a)>=2.2250738585072014e-308)return 0;if(a!=0)return 1;return 2}return isNaN(a)?4:3}function
atO(a){return 0}var
aua=Math.log2&&Math.log2(1.1235582092889474E+307)==1020;function
at$(a){if(aua)return Math.floor(Math.log2(a));var
b=0;if(a==0)return-Infinity;if(a>=1)while(a>=2){a/=2;b++}else
while(a<1){a*=2;b--}return b}function
DS(a){var
b=new
Float32Array(1);b[0]=a;var
c=new
Int32Array(b.buffer);return c[0]|0}var
L6=Math.pow(2,-24);function
Mb(a){throw a}function
Md(){Mb(bg.Division_by_zero)}function
K(a,b,c){this.lo=a&gt;this.mi=b&gt;this.hi=c&co}K.prototype.caml_custom="_j";K.prototype.copy=function(){return new
K(this.lo,this.mi,this.hi)};K.prototype.ucompare=function(a){if(this.hi>a.hi)return 1;if(this.hi<a.hi)return-1;if(this.mi>a.mi)return 1;if(this.mi<a.mi)return-1;if(this.lo>a.lo)return 1;if(this.lo<a.lo)return-1;return 0};K.prototype.compare=function(a){var
b=this.hi<<16,c=a.hi<<16;if(b>c)return 1;if(b<c)return-1;if(this.mi>a.mi)return 1;if(this.mi<a.mi)return-1;if(this.lo>a.lo)return 1;if(this.lo<a.lo)return-1;return 0};K.prototype.neg=function(){var
a=-this.lo,b=-this.mi+(a>>24),c=-this.hi+(b>>24);return new
K(a,b,c)};K.prototype.add=function(a){var
b=this.lo+a.lo,c=this.mi+a.mi+(b>>24),d=this.hi+a.hi+(c>>24);return new
K(b,c,d)};K.prototype.sub=function(a){var
b=this.lo-a.lo,c=this.mi-a.mi+(b>>24),d=this.hi-a.hi+(c>>24);return new
K(b,c,d)};K.prototype.mul=function(a){var
b=this.lo*a.lo,c=(b*L6|0)+this.mi*a.lo+this.lo*a.mi,d=(c*L6|0)+this.hi*a.lo+this.mi*a.mi+this.lo*a.hi;return new
K(b,c,d)};K.prototype.isZero=function(){return(this.lo|this.mi|this.hi)==0};K.prototype.isNeg=function(){return this.hi<<16<0};K.prototype.and=function(a){return new
K(this.lo&a.lo,this.mi&a.mi,this.hi&a.hi)};K.prototype.or=function(a){return new
K(this.lo|a.lo,this.mi|a.mi,this.hi|a.hi)};K.prototype.xor=function(a){return new
K(this.lo^a.lo,this.mi^a.mi,this.hi^a.hi)};K.prototype.shift_left=function(a){a=a&63;if(a==0)return this;if(a<24)return new
K(this.lo<<a,this.mi<<a|this.lo>>24-a,this.hi<<a|this.mi>>24-a);if(a<48)return new
K(0,this.lo<<a-24,this.mi<<a-24|this.lo>>48-a);return new
K(0,0,this.lo<<a-48)};K.prototype.shift_right_unsigned=function(a){a=a&63;if(a==0)return this;if(a<24)return new
K(this.lo>>a|this.mi<<24-a,this.mi>>a|this.hi<<24-a,this.hi>>a);if(a<48)return new
K(this.mi>>a-24|this.hi<<48-a,this.hi>>a-24,0);return new
K(this.hi>>a-48,0,0)};K.prototype.shift_right=function(a){a=a&63;if(a==0)return this;var
c=this.hi<<16>>16;if(a<24)return new
K(this.lo>>a|this.mi<<24-a,this.mi>>a|c<<24-a,this.hi<<16>>a>>>16);var
b=this.hi<<16>>31;if(a<48)return new
K(this.mi>>a-24|this.hi<<48-a,this.hi<<16>>a-24>>16,b&co);return new
K(this.hi<<16>>a-32,b,b)};K.prototype.lsl1=function(){this.hi=this.hi<<1|this.mi>>23;this.mi=(this.mi<<1|this.lo>>23)&gt;this.lo=this.lo<<1&gt};K.prototype.lsr1=function(){this.lo=(this.lo>>>1|this.mi<<23)&gt;this.mi=(this.mi>>>1|this.hi<<23)&gt;this.hi=this.hi>>>1};K.prototype.udivmod=function(a){var
d=0,c=this.copy(),b=a.copy(),e=new
K(0,0,0);while(c.ucompare(b)>0){d++;b.lsl1()}while(d>=0){d--;e.lsl1();if(c.ucompare(b)>=0){e.lo++;c=c.sub(b)}b.lsr1()}return{quotient:e,modulus:c}};K.prototype.div=function(a){var
b=this;if(a.isZero())Md();var
d=b.hi^a.hi;if(b.hi&ck)b=b.neg();if(a.hi&ck)a=a.neg();var
c=b.udivmod(a).quotient;if(d&ck)c=c.neg();return c};K.prototype.mod=function(a){var
b=this;if(a.isZero())Md();var
d=b.hi;if(b.hi&ck)b=b.neg();if(a.hi&ck)a=a.neg();var
c=b.udivmod(a).modulus;if(d&ck)c=c.neg();return c};K.prototype.toInt=function(){return this.lo|this.mi<<24};K.prototype.toFloat=function(){return(this.hi<<16)*Math.pow(2,32)+this.mi*Math.pow(2,24)+this.lo};K.prototype.toArray=function(){return[this.hi>>8,this.hi&ho,this.mi>>16,this.mi>>8&ho,this.mi&ho,this.lo>>16,this.lo>>8&ho,this.lo&ho]};K.prototype.lo32=function(){return this.lo|(this.mi&ho)<<24};K.prototype.hi32=function(){return this.mi>>>8&co|this.hi<<16};function
qc(a,b,c){return new
K(a,b,c)}function
qb(a){if(!isFinite(a)){if(isNaN(a))return qc(1,0,Kh);return a>0?qc(0,0,Kh):qc(0,0,0xfff0)}var
f=a==0&&1/a==-Infinity?ck:a>=0?0:ck;if(f)a=-a;var
b=at$(a)+1023;if(b<=0){b=0;a/=Math.pow(2,-Il)}else{a/=Math.pow(2,b-KQ);if(a<16){a*=2;b-=1}if(b==0)a/=2}var
d=Math.pow(2,24),c=a|0;a=(a-c)*d;var
e=a|0;a=(a-e)*d;var
g=a|0;c=c&BO|f|b<<4;return qc(g,e,c)}function
ld(a){return a.toArray()}function
LW(a,b,c){a.write(32,b.dims.length);a.write(32,b.kind|b.layout<<8);if(b.caml_custom==nx)for(var
d=0;d<b.dims.length;d++)if(b.dims[d]<co)a.write(16,b.dims[d]);else{a.write(16,co);a.write(32,0);a.write(32,b.dims[d])}else
for(var
d=0;d<b.dims.length;d++)a.write(32,b.dims[d]);switch(b.kind){case
2:case
3:case
12:for(var
d=0;d<b.data.length;d++)a.write(8,b.data[d]);break;case
4:case
5:for(var
d=0;d<b.data.length;d++)a.write(16,b.data[d]);break;case
6:for(var
d=0;d<b.data.length;d++)a.write(32,b.data[d]);break;case
8:case
9:a.write(8,0);for(var
d=0;d<b.data.length;d++)a.write(32,b.data[d]);break;case
7:for(var
d=0;d<b.data.length/2;d++){var
f=ld(b.get(d));for(var
e=0;e<8;e++)a.write(8,f[e])}break;case
1:for(var
d=0;d<b.data.length;d++){var
f=ld(qb(b.get(d)));for(var
e=0;e<8;e++)a.write(8,f[e])}break;case
0:for(var
d=0;d<b.data.length;d++){var
f=DS(b.get(d));a.write(32,f)}break;case
10:for(var
d=0;d<b.data.length/2;d++){var
e=b.get(d);a.write(32,DS(e[1]));a.write(32,DS(e[2]))}break;case
11:for(var
d=0;d<b.data.length/2;d++){var
g=b.get(d),f=ld(qb(g[1]));for(var
e=0;e<8;e++)a.write(8,f[e]);var
f=ld(qb(g[2]));for(var
e=0;e<8;e++)a.write(8,f[e])}break}c[0]=(4+b.dims.length)*4;c[1]=(4+b.dims.length)*8}function
LU(a){switch(a){case
7:case
10:case
11:return 2;default:return 1}}function
atq(a,b){var
c;switch(a){case
0:c=Float32Array;break;case
1:c=Float64Array;break;case
2:c=Int8Array;break;case
3:c=Uint8Array;break;case
4:c=Int16Array;break;case
5:c=Uint16Array;break;case
6:c=Int32Array;break;case
7:c=Int32Array;break;case
8:c=Int32Array;break;case
9:c=Int32Array;break;case
10:c=Float32Array;break;case
11:c=Float64Array;break;case
12:c=Uint8Array;break}if(!c)aM("Bigarray.create: unsupported kind");var
d=new
c(b*LU(a));return d}function
DT(a){var
b=new
Int32Array(1);b[0]=a;var
c=new
Float32Array(b.buffer);return c[0]}function
lb(a){return new
K(a[7]<<0|a[6]<<8|a[5]<<16,a[4]<<0|a[3]<<8|a[2]<<16,a[1]<<0|a[0]<<8)}function
DU(a){var
f=a.lo,g=a.mi,c=a.hi,d=(c&0x7fff)>>4;if(d==IE)return(f|g|c&BO)==0?c&ck?-Infinity:Infinity:NaN;var
e=Math.pow(2,-24),b=(f*e+g)*e+(c&BO);if(d>0){b+=16;b*=Math.pow(2,d-KQ)}else
b*=Math.pow(2,-Il);if(c&ck)b=-b;return b}function
DM(a){var
d=a.length,c=1;for(var
b=0;b<d;b++){if(a[b]<0)aM("Bigarray.create: negative dimension");c=c*a[b]}return c}function
atF(a,b){return new
K(a&gt,a>>>24&ho|(b&co)<<8,b>>>16&co)}function
DV(a){return a.hi32()}function
DW(a){return a.lo32()}var
atr=nx;function
hY(a,b,c,d){this.kind=a;this.layout=b;this.dims=c;this.data=d}hY.prototype.caml_custom=atr;hY.prototype.offset=function(a){var
c=0;if(typeof
a==="number")a=[a];if(!(a
instanceof
Array))aM("bigarray.js: invalid offset");if(this.dims.length!=a.length)aM("Bigarray.get/set: bad number of dimensions");if(this.layout==0)for(var
b=0;b<this.dims.length;b++){if(a[b]<0||a[b]>=this.dims[b])k$();c=c*this.dims[b]+a[b]}else
for(var
b=this.dims.length-1;b>=0;b--){if(a[b]<1||a[b]>this.dims[b])k$();c=c*this.dims[b]+(a[b]-1)}return c};hY.prototype.get=function(a){switch(this.kind){case
7:var
d=this.data[a*2+0],b=this.data[a*2+1];return atF(d,b);case
10:case
11:var
e=this.data[a*2+0],c=this.data[a*2+1];return[je,e,c];default:return this.data[a]}};hY.prototype.set=function(a,b){switch(this.kind){case
7:this.data[a*2+0]=DW(b);this.data[a*2+1]=DV(b);break;case
10:case
11:this.data[a*2+0]=b[1];this.data[a*2+1]=b[2];break;default:this.data[a]=b;break}return 0};hY.prototype.fill=function(a){switch(this.kind){case
7:var
c=DW(a),e=DV(a);if(c==e)this.data.fill(c);else
for(var
b=0;b<this.data.length;b++)this.data[b]=b%2==0?c:e;break;case
10:case
11:var
d=a[1],f=a[2];if(d==f)this.data.fill(d);else
for(var
b=0;b<this.data.length;b++)this.data[b]=b%2==0?d:f;break;default:this.data.fill(a);break}};hY.prototype.compare=function(a,b){if(this.layout!=a.layout||this.kind!=a.kind){var
f=this.kind|this.layout<<8,g=a.kind|a.layout<<8;return g-f}if(this.dims.length!=a.dims.length)return a.dims.length-this.dims.length;for(var
c=0;c<this.dims.length;c++)if(this.dims[c]!=a.dims[c])return this.dims[c]<a.dims[c]?-1:1;switch(this.kind){case
0:case
1:case
10:case
11:var
d,e;for(var
c=0;c<this.data.length;c++){d=this.data[c];e=a.data[c];if(d<e)return-1;if(d>e)return 1;if(d!=e){if(!b)return NaN;if(d==d)return 1;if(e==e)return-1}}break;case
7:for(var
c=0;c<this.data.length;c+=2){if(this.data[c+1]<a.data[c+1])return-1;if(this.data[c+1]>a.data[c+1])return 1;if(this.data[c]>>>0<a.data[c]>>>0)return-1;if(this.data[c]>>>0>a.data[c]>>>0)return 1}break;case
2:case
3:case
4:case
5:case
6:case
8:case
9:case
12:for(var
c=0;c<this.data.length;c++){if(this.data[c]<a.data[c])return-1;if(this.data[c]>a.data[c])return 1}break}return 0};function
jx(a,b,c,d){this.kind=a;this.layout=b;this.dims=c;this.data=d}jx.prototype=new
hY();jx.prototype.offset=function(a){if(typeof
a!=="number")if(a
instanceof
Array&&a.length==1)a=a[0];else
aM("Ml_Bigarray_c_1_1.offset");if(a<0||a>=this.dims[0])k$();return a};jx.prototype.get=function(a){return this.data[a]};jx.prototype.set=function(a,b){this.data[a]=b;return 0};jx.prototype.fill=function(a){this.data.fill(a);return 0};function
LS(a,b,c,d){var
e=LU(a);if(DM(c)*e!=d.length)aM("length doesn't match dims");if(b==0&&c.length==1&&e==1)return new
jx(a,b,c,d);return new
hY(a,b,c,d)}function
bf(a){if(!bg.Failure)bg.Failure=[c,cw(xu),-3];D4(bg.Failure,a)}function
LT(a,b,c){var
k=a.read32s();if(k<0||k>16)bf("input_value: wrong number of bigarray dimensions");var
r=a.read32s(),l=r&ho,q=r>>8&1,j=[];if(c==nx)for(var
d=0;d<k;d++){var
p=a.read16u();if(p==co){var
u=a.read32u(),v=a.read32u();if(u!=0)bf("input_value: bigarray dimension overflow in 32bit");p=v}j.push(p)}else
for(var
d=0;d<k;d++)j.push(a.read32u());var
f=DM(j),h=atq(l,f),i=LS(l,q,j,h);switch(l){case
2:for(var
d=0;d<f;d++)h[d]=a.read8s();break;case
3:case
12:for(var
d=0;d<f;d++)h[d]=a.read8u();break;case
4:for(var
d=0;d<f;d++)h[d]=a.read16s();break;case
5:for(var
d=0;d<f;d++)h[d]=a.read16u();break;case
6:for(var
d=0;d<f;d++)h[d]=a.read32s();break;case
8:case
9:var
t=a.read8u();if(t)bf("input_value: cannot read bigarray with 64-bit OCaml ints");for(var
d=0;d<f;d++)h[d]=a.read32s();break;case
7:var
g=new
Array(8);for(var
d=0;d<f;d++){for(var
e=0;e<8;e++)g[e]=a.read8u();var
s=lb(g);i.set(d,s)}break;case
1:var
g=new
Array(8);for(var
d=0;d<f;d++){for(var
e=0;e<8;e++)g[e]=a.read8u();var
m=DU(lb(g));i.set(d,m)}break;case
0:for(var
d=0;d<f;d++){var
m=DT(a.read32s());i.set(d,m)}break;case
10:for(var
d=0;d<f;d++){var
o=DT(a.read32s()),n=DT(a.read32s());i.set(d,[je,o,n])}break;case
11:var
g=new
Array(8);for(var
d=0;d<f;d++){for(var
e=0;e<8;e++)g[e]=a.read8u();var
o=DU(lb(g));for(var
e=0;e<8;e++)g[e]=a.read8u();var
n=DU(lb(g));i.set(d,[je,o,n])}break}b[0]=(4+k)*4;return LS(l,q,j,h)}function
LR(a,b,c){return a.compare(b,c)}function
L9(a,b){return Math.imul(a,b)}function
hv(a,b){b=L9(b,0xcc9e2d51|0);b=b<<15|b>>>32-15;b=L9(b,0x1b873593);a^=b;a=a<<13|a>>>32-13;return(a+(a<<2)|0)+(0xe6546b64|0)|0}function
atA(a,b){a=hv(a,DW(b));a=hv(a,DV(b));return a}function
L5(a,b){return atA(a,qb(b))}function
LV(a){var
c=DM(a.dims),d=0;switch(a.kind){case
2:case
3:case
12:if(c>kS)c=kS;var
e=0,b=0;for(b=0;b+4<=a.data.length;b+=4){e=a.data[b+0]|a.data[b+1]<<8|a.data[b+2]<<16|a.data[b+3]<<24;d=hv(d,e)}e=0;switch(c&3){case
3:e=a.data[b+2]<<16;case
2:e|=a.data[b+1]<<8;case
1:e|=a.data[b+0];d=hv(d,e)}break;case
4:case
5:if(c>an)c=an;var
e=0,b=0;for(b=0;b+2<=a.data.length;b+=2){e=a.data[b+0]|a.data[b+1]<<16;d=hv(d,e)}if((c&1)!=0)d=hv(d,a.data[b]);break;case
6:if(c>64)c=64;for(var
b=0;b<c;b++)d=hv(d,a.data[b]);break;case
8:case
9:if(c>64)c=64;for(var
b=0;b<c;b++)d=hv(d,a.data[b]);break;case
7:if(c>32)c=32;c*=2;for(var
b=0;b<c;b++)d=hv(d,a.data[b]);break;case
10:c*=2;case
0:if(c>64)c=64;for(var
b=0;b<c;b++)d=L5(d,a.data[b]);break;case
11:c*=2;case
1:if(c>32)c=32;for(var
b=0;b<c;b++)d=L5(d,a.data[b]);break}return d}function
atC(a,b){b[0]=4;return a.read32s()}function
at0(a,b){switch(a.read8u()){case
1:b[0]=4;return a.read32s();case
2:bf("input_value: native integer value too large");default:bf("input_value: ill-formed native integer")}}function
atN(a,b){var
d=new
Array(8);for(var
c=0;c<8;c++)d[c]=a.read8u();b[0]=8;return lb(d)}function
atK(a,b,c){var
e=ld(b);for(var
d=0;d<8;d++)a.write(8,e[d]);c[0]=8;c[1]=8}function
atE(a,b,c){return a.compare(b)}function
atH(a){return a.lo32()^a.hi32()}var
L1={"_j":{deserialize:atN,serialize:atK,fixed_length:8,compare:atE,hash:atH},"_i":{deserialize:atC,fixed_length:4},"_n":{deserialize:at0,fixed_length:4},"_bigarray":{deserialize:function(a,b){return LT(a,b,"_bigarray")},serialize:LW,compare:LR,hash:LV},"_bigarr02":{deserialize:function(a,b){return LT(a,b,nx)},serialize:LW,compare:LR,hash:LV}};function
DO(a){return L1[a.caml_custom]&&L1[a.caml_custom].compare}function
LZ(a,b,c,d){var
f=DO(b);if(f){var
e=c>0?f(b,a,d):f(a,b,d);if(d&&e!=e)return c;if(+e!=+e)return+e;if((e|0)!=0)return e|0}return c}function
D0(a){return typeof
a==="string"&&!/[^\x00-\xff]/.test(a)}function
DZ(a){return a
instanceof
hr}function
L0(a){if(typeof
a==="number")return kZ;else if(DZ(a))return jr;else if(D0(a))return 1252;else if(a
instanceof
Array&&a[0]===a[0]>>>0&&a[0]<=kT){var
b=a[0]|0;return b==je?0:b}else if(a
instanceof
String)return LK;else if(typeof
a=="string")return LK;else if(a
instanceof
Number)return kZ;else if(a&&a.caml_custom)return vG;else if(a&&a.compare)return 1256;else if(typeof
a=="function")return 1247;else if(typeof
a=="symbol")return 1251;return 1001}function
cu(a,b){if(a<b)return-1;if(a==b)return 0;return 1}function
z(a,b){return a<b?-1:a>b?1:0}function
atu(a,b){a.t&6&&p_(a);b.t&6&&p_(b);return a.c<b.c?-1:a.c>b.c?1:0}function
p8(a,b,c){var
f=[];for(;;){if(!(c&&a===b)){var
e=L0(a);if(e==im){a=a[1];continue}var
g=L0(b);if(g==im){b=b[1];continue}if(e!==g){if(e==kZ){if(g==vG)return LZ(a,b,-1,c);return-1}if(g==kZ){if(e==vG)return LZ(b,a,1,c);return 1}return e<g?-1:1}switch(e){case
247:aM(yZ);break;case
248:var
d=cu(a[2],b[2]);if(d!=0)return d|0;break;case
249:aM(yZ);break;case
250:aM("equal: got Forward_tag, should not happen");break;case
251:aM("equal: abstract value");break;case
252:if(a!==b){var
d=atu(a,b);if(d!=0)return d|0}break;case
253:aM("equal: got Double_tag, should not happen");break;case
254:aM("equal: got Double_array_tag, should not happen");break;case
255:aM("equal: got Custom_tag, should not happen");break;case
1247:aM(yZ);break;case
1255:var
i=DO(a);if(i!=DO(b))return a.caml_custom<b.caml_custom?-1:1;if(!i)aM("compare: abstract value");var
d=i(a,b,c);if(d!=d)return c?-1:d;if(d!==(d|0))return-1;if(d!=0)return d|0;break;case
1256:var
d=a.compare(b,c);if(d!=d)return c?-1:d;if(d!==(d|0))return-1;if(d!=0)return d|0;break;case
1000:a=+a;b=+b;if(a<b)return-1;if(a>b)return 1;if(a!=b){if(!c)return NaN;if(a==a)return 1;if(b==b)return-1}break;case
1001:if(a<b)return-1;if(a>b)return 1;if(a!=b){if(!c)return NaN;if(a==a)return 1;if(b==b)return-1}break;case
1251:if(a!==b){if(!c)return NaN;return 1}break;case
1252:var
a=hw(a),b=hw(b);if(a!==b){if(a<b)return-1;if(a>b)return 1}break;case
12520:var
a=a.toString(),b=b.toString();if(a!==b){if(a<b)return-1;if(a>b)return 1}break;case
246:case
254:default:if(atO(e)){aM("compare: continuation value");break}if(a.length!=b.length)return a.length<b.length?-1:1;if(a.length>1)f.push(a,b,1);break}}if(f.length==0)return 0;var
h=f.pop();b=f.pop();a=f.pop();if(h+1<a.length)f.push(a,b,h+1);a=a[h];b=b[h]}}function
LY(a,b){return p8(a,b,true)}function
atw(){return[0]}function
au(a){if(a<0)aM("Bytes.create");return new
hr(a?2:9,w,a)}function
p$(a,b){return+(p8(a,b,false)==0)}function
aty(a,b,c,d){if(c>0)if(b==0&&(c>=a.l||a.t==2&&c>=a.c.length))if(d==0){a.c=w;a.t=2}else{a.c=jA(c,String.fromCharCode(d));a.t=c==a.l?0:2}else{if(a.t!=4)p9(a);for(c+=b;b<c;b++)a.c[b]=d}return 0}function
DQ(a){var
b;a=hw(a);b=+a;if(a.length>0&&b===b)return b;a=a.replace(/_/g,w);b=+a;if(a.length>0&&b===b||/^[+-]?nan$/i.test(a))return b;var
c=/^ *([+-]?)0x([0-9a-f]+)\.?([0-9a-f]*)(p([+-]?[0-9]+))?/i.exec(a);if(c){var
d=c[3].replace(/0+$/,w),f=parseInt(c[1]+c[2]+d,16),e=(c[5]|0)-4*d.length;b=f*Math.pow(2,e);return b}if(/^\+?inf(inity)?$/i.test(a))return Infinity;if(/^-inf(inity)?$/i.test(a))return-Infinity;bf("float_of_string")}function
D3(a){a=hw(a);var
e=a.length;if(e>31)aM("format_int: format too long");var
b={justify:hi,signstyle:ee,filler:b7,alternate:false,base:0,signedconv:false,width:0,uppercase:false,sign:1,prec:-1,conv:La};for(var
d=0;d<e;d++){var
c=a.charAt(d);switch(c){case"-":b.justify=ee;break;case"+":case" ":b.signstyle=c;break;case"0":b.filler=bn;break;case"#":b.alternate=true;break;case"1":case"2":case"3":case"4":case"5":case"6":case"7":case"8":case"9":b.width=0;while(c=a.charCodeAt(d)-48,c>=0&&c<=9){b.width=b.width*10+c;d++}d--;break;case".":b.prec=0;d++;while(c=a.charCodeAt(d)-48,c>=0&&c<=9){b.prec=b.prec*10+c;d++}d--;case"d":case"i":b.signedconv=true;case"u":b.base=10;break;case"x":b.base=16;break;case"X":b.base=16;b.uppercase=true;break;case"o":b.base=8;break;case"e":case"f":case"g":b.signedconv=true;b.conv=c;break;case"E":case"F":case"G":b.signedconv=true;b.uppercase=true;b.conv=c.toLowerCase();break}}return b}function
DP(a,b){if(a.uppercase)b=b.toUpperCase();var
e=b.length;if(a.signedconv&&(a.sign<0||a.signstyle!=ee))e++;if(a.alternate){if(a.base==8)e+=1;if(a.base==16)e+=2}var
c=w;if(a.justify==hi&&a.filler==b7)for(var
d=e;d<a.width;d++)c+=b7;if(a.signedconv)if(a.sign<0)c+=ee;else if(a.signstyle!=ee)c+=a.signstyle;if(a.alternate&&a.base==8)c+=bn;if(a.alternate&&a.base==16)c+=a.uppercase?"0X":iz;if(a.justify==hi&&a.filler==bn)for(var
d=e;d<a.width;d++)c+=bn;c+=b;if(a.justify==ee)for(var
d=e;d<a.width;d++)c+=b7;return cw(c)}function
DR(a,b){function
j(a,b){if(Math.abs(a)<1.0)return a.toFixed(b);else{var
c=parseInt(a.toString().split(hi)[1]);if(c>20){c-=20;a/=Math.pow(10,c);a+=new
Array(c+1).join(bn);if(b>0)a=a+cT+new
Array(b+1).join(bn);return a}else
return a.toFixed(b)}}var
c,f=D3(a),e=f.prec<0?6:f.prec;if(b<0||b==0&&1/b==-Infinity){f.sign=-1;b=-b}if(isNaN(b)){c=AT;f.filler=b7}else if(!isFinite(b)){c="inf";f.filler=b7}else
switch(f.conv){case"e":var
c=b.toExponential(e),d=c.length;if(c.charAt(d-3)==uX)c=c.slice(0,d-1)+bn+c.slice(d-1);break;case"f":c=j(b,e);break;case"g":e=e?e:1;c=b.toExponential(e-1);var
i=c.indexOf(uX),h=+c.slice(i+1);if(h<-4||b>=1e21||b.toFixed(0).length>e){var
d=i-1;while(c.charAt(d)==bn)d--;if(c.charAt(d)==cT)d--;c=c.slice(0,d+1)+c.slice(i);d=c.length;if(c.charAt(d-3)==uX)c=c.slice(0,d-1)+bn+c.slice(d-1);break}else{var
g=e;if(h<0){g-=h+1;c=b.toFixed(g)}else
while(c=b.toFixed(g),c.length>e+1)g--;if(g){var
d=c.length-1;while(c.charAt(d)==bn)d--;if(c.charAt(d)==cT)d--;c=c.slice(0,d+1)}}break}return DP(f,c)}function
qa(a,b){if(hw(a)==kV)return cw(w+b);var
c=D3(a);if(b<0)if(c.signedconv){c.sign=-1;b=-b}else
b>>>=0;var
d=b.toString(c.base);if(c.prec>=0){c.filler=b7;var
e=c.prec-d.length;if(e>0)d=jA(e,bn)+d}return DP(c,d)}var
L$=0;function
hu(){return L$++}function
L3(){return[0]}var
qf=[];function
I(a,b,c){var
e=a[1],i=qf[c];if(i===undefined)for(var
h=qf.length;h<c;h++)qf[h]=0;else if(e[i]===b)return e[i-1];var
d=3,g=e[1]*2+1,f;while(d<g){f=d+g>>1|1;if(b<e[f+1])g=f-2;else
d=f}qf[c]=d+1;return b==e[d+1]?e[d]:0}function
at_(a){for(var
f=w,c=f,b,h,d=0,g=a.length;d<g;d++){b=a.charCodeAt(d);if(b<cs){for(var
e=d+1;e<g&&(b=a.charCodeAt(e))<cs;e++);if(e-d>xr){c.substr(0,1);f+=c;c=w;f+=a.slice(d,e)}else
c+=a.slice(d,e);if(e==g)break;d=e}if(b<JZ){c+=String.fromCharCode(0xc0|b>>6);c+=String.fromCharCode(cs|b&jm)}else if(b<0xd800||b>=K5)c+=String.fromCharCode(Jk|b>>12,cs|b>>6&jm,cs|b&jm);else if(b>=0xdbff||d+1==g||(h=a.charCodeAt(d+1))<Ir||h>K5)c+="\xef\xbf\xbd";else{d++;b=(b<<10)+h-0x35fdc00;c+=String.fromCharCode(J6|b>>18,cs|b>>12&jm,cs|b>>6&jm,cs|b&jm)}if(c.length>kK){c.substr(0,1);f+=c;c=w}}return f+c}function
b$(a){return D8(a)?cw(a):cw(at_(a))}function
atB(a,b,c){if(!isFinite(a)){if(isNaN(a))return b$(AT);return b$(a>0?Kz:"-infinity")}var
j=a==0&&1/a==-Infinity?1:a>=0?0:1;if(j)a=-a;var
e=0;if(a==0);else if(a<1)while(a<1&&e>-1022){a*=2;e--}else
while(a>=2){a/=2;e++}var
k=e<0?w:hi,f=w;if(j)f=ee;else
switch(c){case
43:f=hi;break;case
32:f=b7;break;default:break}if(b>=0&&b<13){var
h=Math.pow(2,b*4);a=Math.round(a*h)/h}var
d=a.toString(16);if(b>=0){var
i=d.indexOf(cT);if(i<0)d+=cT+jA(b,bn);else{var
g=i+1+b;if(d.length<g)d+=jA(g-d.length,bn);else
d=d.substr(0,g)}}return b$(f+iz+d+"p"+k+e.toString(10))}function
atJ(a){return+a.isZero()}function
qd(a){return new
K(a&gt,a>>24&gt,a>>31&co)}function
atM(a){return a.toInt()}function
atI(a){return+a.isNeg()}function
DX(a){return a.neg()}function
atG(a,b){var
c=D3(a);if(c.signedconv&&atI(b)){c.sign=-1;b=DX(b)}var
d=w,h=qd(c.base),g="0123456789abcdef";do{var
f=b.udivmod(h);b=f.quotient;d=g.charAt(atM(f.modulus))+d}while(!atJ(b));if(c.prec>=0){c.filler=b7;var
e=c.prec-d.length;if(e>0)d=jA(e,bn)+d}return DP(c,d)}function
J(a){return a.length}function
u(a,b){return a.charCodeAt(b)}function
atD(a,b){return a.add(b)}function
atL(a,b){return a.mul(b)}function
DY(a,b){return a.ucompare(b)<0}function
Ma(a){var
b=0,e=J(a),c=10,d=1;if(e>0)switch(u(a,b)){case
45:b++;d=-1;break;case
43:b++;d=1;break}if(b+1<e&&u(a,b)==48)switch(u(a,b+1)){case
120:case
88:c=16;b+=2;break;case
111:case
79:c=8;b+=2;break;case
98:case
66:c=2;b+=2;break;case
117:case
85:b+=2;break}return[b,d,c]}function
qg(a){if(a>=48&&a<=57)return a-48;if(a>=65&&a<=90)return a-55;if(a>=97&&a<=ad)return a-87;return-1}function
lc(a){var
g=Ma(a),f=g[0],i=g[1],d=g[2],h=qd(d),j=new
K(gt,0xfffffff,co).udivmod(h).quotient,e=u(a,f),b=qg(e);if(b<0||b>=d)bf(hn);var
c=qd(b);for(;;){f++;e=u(a,f);if(e==95)continue;b=qg(e);if(b<0||b>=d)break;if(DY(j,c))bf(hn);b=qd(b);c=atD(atL(h,c),b);if(DY(c,b))bf(hn)}if(f!=J(a))bf(hn);if(d==10&&DY(new
K(0,0,ck),c))bf(hn);if(i<0)c=DX(c);return c}function
qe(a){return a.toFloat()}function
cv(a){var
h=Ma(a),d=h[0],i=h[1],e=h[2],g=J(a),j=-1>>>0,f=d<g?u(a,d):0,c=qg(f);if(c<0||c>=e)bf(hn);var
b=c;for(d++;d<g;d++){f=u(a,d);if(f==95)continue;c=qg(f);if(c<0||c>=e)break;b=e*b+c;if(b>j)bf(hn)}if(d!=g)bf(hn);b=i*b;if(e==10&&(b|0)!=b)bf(hn);return b|0}function
atQ(a){return a.slice(1)}function
H(a){if(D8(a))return a;return Mh(a)}function
atS(a){var
c={};for(var
b=1;b<a.length;b++){var
d=a[b];c[H(d[1])]=d[2]}return c}var
p7=ht;function
atR(a){return a.l>=0?a.l:a.l=a.length}function
atT(a){return function(){var
d=atR(a),c=new
Array(d);for(var
b=0;b<d;b++)c[b]=arguments[b];return p7(a,c)}}function
atU(a,b){return+(p8(a,b,false)<0)}function
jz(a,b){if(a<0)k$();var
a=a+1|0,c=new
Array(a);c[0]=0;for(var
d=1;d<a;d++)c[d]=b;return c}function
bV(a){return a.l}function
atW(){return 0}function
Y(a){D4(bg.Sys_error,a)}var
hx=new
Array();function
g5(a){var
b=hx[a];if(!b.opened)Y("Cannot flush a closed channel");if(!b.buffer||b.buffer_curr==0)return 0;if(b.output)b.output(qi(b.buffer,0,b.buffer_curr));else
b.file.write(b.offset,b.buffer,0,b.buffer_curr);b.offset+=b.buffer_curr;b.buffer_curr=0;return 0}function
LQ(){}function
aug(a){var
d=J(a),c=new
Array(d),b=0;for(;b<d;b++)c[b]=u(a,b);return c}function
Mg(a){if(a.t!=4)p9(a);return a.c}function
ct(a,b){this.fs=require(zM);this.fd=a;this.flags=b}ct.prototype=new
LQ();ct.prototype.constructor=ct;ct.prototype.truncate=function(a){try{this.fs.ftruncateSync(this.fd,a|0)}catch(f){Y(f.toString())}};ct.prototype.length=function(){try{return this.fs.fstatSync(this.fd).size}catch(f){Y(f.toString())}};ct.prototype.write=function(a,b,c,d){try{if(this.flags.isCharacterDevice)this.fs.writeSync(this.fd,b,c,d);else
this.fs.writeSync(this.fd,b,c,d,a)}catch(f){Y(f.toString())}return 0};ct.prototype.read=function(a,b,c,d){try{if(this.flags.isCharacterDevice)var
e=this.fs.readSync(this.fd,b,c,d);else
var
e=this.fs.readSync(this.fd,b,c,d,a);return e}catch(f){Y(f.toString())}};ct.prototype.close=function(){try{this.fs.closeSync(this.fd);return 0}catch(f){Y(f.toString())}};function
at9(a,b){if(b.name)try{var
d=require(zM),c=d.openSync(b.name,"rs");return new
ct(c,b)}catch(f){}return new
ct(a,b)}var
qj=new
Array(3);function
lf(){return typeof
ak.process!=="undefined"&&typeof
ak.process.versions!=="undefined"&&typeof
ak.process.versions.node!=="undefined"}function
aub(){function
a(a){if(a.charAt(0)===bq)return[w,a.substring(1)];return}function
b(a){var
g=/^([a-zA-Z]:|[\\/]{2}[^\\/]+[\\/]+[^\\/]+)?([\\/])?([\s\S]*?)$/,b=g.exec(a),c=b[1]||w,e=Boolean(c&&c.charAt(1)!==ir);if(Boolean(b[2]||e)){var
d=b[1]||w,f=b[2]||w;return[d,a.substring(d.length+f.length)]}return}return lf()&&ak.process&&ak.process.platform?ak.process.platform===Ls?b:a:a}var
D9=aub();function
Mf(a){return a.slice(-1)!==bq?a+bq:a}if(lf()&&ak.process&&ak.process.cwd)var
la=ak.process.cwd().replace(/\\/g,bq);else
var
la="/static";la=Mf(la);function
atV(a){a=H(a);if(!D9(a))a=la+a;var
e=D9(a),d=e[1].split(bq),b=[];for(var
c=0;c<d.length;c++)switch(d[c]){case"..":if(b.length>1)b.pop();break;case".":break;case"":break;default:b.push(d[c]);break}b.unshift(e[0]);b.orig=a;return b}var
aue=["E2BIG","EACCES","EAGAIN",CG,"EBUSY","ECHILD","EDEADLK","EDOM",Iw,"EFAULT","EFBIG","EINTR","EINVAL","EIO","EISDIR","EMFILE","EMLINK","ENAMETOOLONG","ENFILE","ENODEV",wU,"ENOEXEC","ENOLCK","ENOMEM","ENOSPC","ENOSYS",xN,LM,"ENOTTY","ENXIO","EPERM","EPIPE","ERANGE","EROFS","ESPIPE","ESRCH","EXDEV","EWOULDBLOCK","EINPROGRESS","EALREADY","ENOTSOCK","EDESTADDRREQ","EMSGSIZE","EPROTOTYPE","ENOPROTOOPT","EPROTONOSUPPORT","ESOCKTNOSUPPORT","EOPNOTSUPP","EPFNOSUPPORT","EAFNOSUPPORT","EADDRINUSE","EADDRNOTAVAIL","ENETDOWN","ENETUNREACH","ENETRESET","ECONNABORTED","ECONNRESET","ENOBUFS","EISCONN","ENOTCONN","ESHUTDOWN","ETOOMANYREFS","ETIMEDOUT","ECONNREFUSED","EHOSTDOWN","EHOSTUNREACH","ELOOP","EOVERFLOW"];function
hz(a,b,c,d){var
e=aue.indexOf(a);if(e<0){if(d==null)d=-9999;e=[0,d]}var
f=[e,b$(b||w),b$(c||w)];return f}var
L_={};function
hZ(a){return L_[a]}function
hy(a,b){throw s([0,a].concat(b))}function
DN(a){if(!(a
instanceof
Uint8Array))a=new
Uint8Array(a);return new
hr(4,a,a.length)}function
Mc(a){Y(a+pY)}function
br(a){this.data=a}br.prototype=new
LQ();br.prototype.constructor=br;br.prototype.truncate=function(a){var
b=this.data;this.data=au(a|0);hs(b,0,this.data,0,a)};br.prototype.length=function(){return bV(this.data)};br.prototype.write=function(a,b,c,d){var
e=this.length();if(a+d>=e){var
f=au(a+d),g=this.data;this.data=f;hs(g,0,this.data,0,e)}hs(DN(b),c,this.data,a,d);return 0};br.prototype.read=function(a,b,c,d){var
e=this.length();if(a+d>=e)d=e-a;if(d){var
f=au(d|0);hs(this.data,a,f,0,d);b.set(Mg(f),c)}return d};function
iD(a,b,c){this.file=b;this.name=a;this.flags=c}iD.prototype.err_closed=function(){Y(this.name+Ly)};iD.prototype.length=function(){if(this.file)return this.file.length();this.err_closed()};iD.prototype.write=function(a,b,c,d){if(this.file)return this.file.write(a,b,c,d);this.err_closed()};iD.prototype.read=function(a,b,c,d){if(this.file)return this.file.read(a,b,c,d);this.err_closed()};iD.prototype.close=function(){this.file=undefined};function
a0(a,b){this.content={};this.root=a;this.lookupFun=b}a0.prototype.nm=function(a){return this.root+a};a0.prototype.create_dir_if_needed=function(a){var
d=a.split(bq),c=w;for(var
b=0;b<d.length-1;b++){c+=d[b]+bq;if(this.content[c])continue;this.content[c]=Symbol("directory")}};a0.prototype.slash=function(a){return/\/$/.test(a)?a:a+bq};a0.prototype.lookup=function(a){if(!this.content[a]&&this.lookupFun){var
b=this.lookupFun(cw(this.root),cw(a));if(b!==0){this.create_dir_if_needed(a);this.content[a]=new
br(g4(b[1]))}}};a0.prototype.exists=function(a){if(a==w)return 1;var
b=this.slash(a);if(this.content[b])return 1;this.lookup(a);return this.content[a]?1:0};a0.prototype.isFile=function(a){return this.exists(a)&&!this.is_dir(a)?1:0};a0.prototype.mkdir=function(a,b,c){var
e=c&&hZ(of);if(this.exists(a))if(e)hy(e,hz(Iw,yK,this.nm(a)));else
Y(a+": File exists");var
d=/^(.*)\/[^/]+/.exec(a);d=d&&d[1]||w;if(!this.exists(d))if(e)hy(e,hz(wU,yK,this.nm(d)));else
Y(d+pY);if(!this.is_dir(d))if(e)hy(e,hz(xN,yK,this.nm(d)));else
Y(d+AQ);this.create_dir_if_needed(this.slash(a))};a0.prototype.rmdir=function(a,b){var
c=b&&hZ(of),d=a==w?w:this.slash(a),f=new
RegExp(n3+d+Id);if(!this.exists(a))if(c)hy(c,hz(wU,Dq,this.nm(a)));else
Y(a+pY);if(!this.is_dir(a))if(c)hy(c,hz(xN,Dq,this.nm(a)));else
Y(a+AQ);for(var
e
in
this.content)if(e.match(f))if(c)hy(c,hz(LM,Dq,this.nm(a)));else
Y(this.nm(a)+": Directory not empty");delete
this.content[d]};a0.prototype.readdir=function(a){var
f=a==w?w:this.slash(a);if(!this.exists(a))Y(a+pY);if(!this.is_dir(a))Y(a+AQ);var
g=new
RegExp(n3+f+Id),d={},c=[];for(var
e
in
this.content){var
b=e.match(g);if(b&&!d[b[1]]){d[b[1]]=true;c.push(b[1])}}return c};a0.prototype.opendir=function(a,b){var
c=b&&hZ(of),d=this.readdir(a),e=false,f=0;return{readSync:function(){if(e)if(c)hy(c,hz(CG,Ko,this.nm(a)));else
Y(a+Jf);if(f==d.length)return null;var
b=d[f];f++;return{name:b}},closeSync:function(){if(e)if(c)hy(c,hz(CG,Ko,this.nm(a)));else
Y(a+Jf);e=true;d=[]}}};a0.prototype.is_dir=function(a){if(a==w)return true;var
b=this.slash(a);return this.content[b]?1:0};a0.prototype.unlink=function(a){var
b=this.content[a]?true:false;delete
this.content[a];return b};a0.prototype.open=function(a,b){var
c;if(b.rdonly&&b.wronly)Y(this.nm(a)+zK);if(b.text&&b.binary)Y(this.nm(a)+sz);this.lookup(a);if(this.content[a]){if(this.is_dir(a))Y(this.nm(a)+Ki);if(b.create&&b.excl)Y(this.nm(a)+yE);c=this.content[a];if(b.truncate)c.truncate()}else if(b.create){this.create_dir_if_needed(a);this.content[a]=new
br(au(0));c=this.content[a]}else
Mc(this.nm(a));return new
iD(this.nm(a),c,b)};a0.prototype.open=function(a,b){var
c;if(b.rdonly&&b.wronly)Y(this.nm(a)+zK);if(b.text&&b.binary)Y(this.nm(a)+sz);this.lookup(a);if(this.content[a]){if(this.is_dir(a))Y(this.nm(a)+Ki);if(b.create&&b.excl)Y(this.nm(a)+yE);c=this.content[a];if(b.truncate)c.truncate()}else if(b.create){this.create_dir_if_needed(a);this.content[a]=new
br(au(0));c=this.content[a]}else
Mc(this.nm(a));return new
iD(this.nm(a),c,b)};a0.prototype.register=function(a,b){var
c;if(this.content[a])Y(this.nm(a)+yE);if(DZ(b))c=new
br(b);if(D0(b))c=new
br(g4(b));else if(b
instanceof
Array)c=new
br(DN(b));else if(typeof
b==="string")c=new
br(LX(b));else if(b.toString){var
d=g4(b$(b.toString()));c=new
br(d)}if(c){this.create_dir_if_needed(a);this.content[a]=c}else
Y(this.nm(a)+" : registering file with invalid content type")};a0.prototype.constructor=a0;function
aL(a){this.fs=require(zM);this.root=a}aL.prototype.nm=function(a){return this.root+a};aL.prototype.exists=function(a){try{return this.fs.existsSync(this.nm(a))?1:0}catch(f){return 0}};aL.prototype.isFile=function(a){try{return this.fs.statSync(this.nm(a)).isFile()?1:0}catch(f){Y(f.toString())}};aL.prototype.mkdir=function(a,b,c){try{this.fs.mkdirSync(this.nm(a),{mode:b});return 0}catch(f){this.raise_nodejs_error(f,c)}};aL.prototype.rmdir=function(a,b){try{this.fs.rmdirSync(this.nm(a));return 0}catch(f){this.raise_nodejs_error(f,b)}};aL.prototype.readdir=function(a,b){try{return this.fs.readdirSync(this.nm(a))}catch(f){this.raise_nodejs_error(f,b)}};aL.prototype.is_dir=function(a){try{return this.fs.statSync(this.nm(a)).isDirectory()?1:0}catch(f){Y(f.toString())}};aL.prototype.unlink=function(a,b){try{var
c=this.fs.existsSync(this.nm(a))?1:0;this.fs.unlinkSync(this.nm(a));return c}catch(f){this.raise_nodejs_error(f,b)}};aL.prototype.open=function(a,b,c){var
d=require("constants"),e=0;for(var
h
in
b)switch(h){case"rdonly":e|=d.O_RDONLY;break;case"wronly":e|=d.O_WRONLY;break;case"append":e|=d.O_WRONLY|d.O_APPEND;break;case"create":e|=d.O_CREAT;break;case"truncate":e|=d.O_TRUNC;break;case"excl":e|=d.O_EXCL;break;case"binary":e|=d.O_BINARY;break;case"text":e|=d.O_TEXT;break;case"nonblock":e|=d.O_NONBLOCK;break}try{var
f=this.fs.openSync(this.nm(a),e),g=this.fs.lstatSync(this.nm(a)).isCharacterDevice();b.isCharacterDevice=g;return new
ct(f,b)}catch(f){this.raise_nodejs_error(f,c)}};aL.prototype.rename=function(a,b,c){try{this.fs.renameSync(this.nm(a),this.nm(b))}catch(f){this.raise_nodejs_error(f,c)}};aL.prototype.stat=function(a,b){try{var
c=this.fs.statSync(this.nm(a));return this.stats_from_js(c)}catch(f){this.raise_nodejs_error(f,b)}};aL.prototype.lstat=function(a,b){try{var
c=this.fs.lstatSync(this.nm(a));return this.stats_from_js(c)}catch(f){this.raise_nodejs_error(f,b)}};aL.prototype.symlink=function(a,b,c,d){try{this.fs.symlinkSync(this.nm(b),this.nm(c),a?"dir":"file");return 0}catch(f){this.raise_nodejs_error(f,d)}};aL.prototype.readlink=function(a,b){try{var
c=this.fs.readlinkSync(this.nm(a),"utf8");return b$(c)}catch(f){this.raise_nodejs_error(f,b)}};aL.prototype.opendir=function(a,b){try{return this.fs.opendirSync(this.nm(a))}catch(f){this.raise_nodejs_error(f,b)}};aL.prototype.raise_nodejs_error=function(a,b){var
c=hZ(of);if(b&&c){var
d=hz(a.code,a.syscall,a.path,a.errno);hy(c,d)}else
Y(a.toString())};aL.prototype.stats_from_js=function(a){var
b;if(a.isFile())b=0;else if(a.isDirectory())b=1;else if(a.isCharacterDevice())b=2;else if(a.isBlockDevice())b=3;else if(a.isSymbolicLink())b=4;else if(a.isFIFO())b=5;else if(a.isSocket())b=6;return[0,a.dev,a.ino,b,a.mode,a.nlink,a.uid,a.gid,a.rdev,a.size,a.atimeMs,a.mtimeMs,a.ctimeMs]};aL.prototype.constructor=aL;function
L4(a){var
b=D9(a);if(!b)return;return b[0]+bq}var
qh=L4(la)||bf("unable to compute caml_root"),jB=[];if(lf())jB.push({path:qh,device:new
aL(qh)});else
jB.push({path:qh,device:new
a0(qh)});jB.push({path:Ie,device:new
a0(Ie)});function
aud(a){var
g=atV(a),a=g.join(bq),f=Mf(a),c;for(var
e=0;e<jB.length;e++){var
b=jB[e];if(f.search(b.path)==0&&(!c||c.path.length<b.path.length))c={path:b.path,device:b.device,rest:a.substring(b.path.length,a.length)}}if(!c&&lf()){var
d=L4(a);if(d&&d.match(/^[a-zA-Z]:\/$/)){var
b={path:d,device:new
aL(d)};jB.push(b);c={path:b.path,device:b.device,rest:a.substring(b.path.length,a.length)}}}if(c)return c;Y("no device found for "+f)}function
k_(a,b){br.call(this,au(0));this.log=function(a){return 0};if(a==1&&typeof
console.log=="function")this.log=console.log;else if(a==2&&typeof
console.error=="function")this.log=console.error;else if(typeof
console.log=="function")this.log=console.log;this.flags=b}k_.prototype.length=function(){return 0};k_.prototype.write=function(a,b,c,d){if(this.log){if(d>0&&c>=0&&c+d<=b.length&&b[c+d-1]==10)d--;var
e=au(d);hs(DN(b),c,e,0,d);this.log(e.toUtf16());return 0}Y(this.fd+Ly)};k_.prototype.read=function(a,b,c,d){Y(this.fd+": file descriptor is write only")};k_.prototype.close=function(){this.log=undefined};function
qk(a,b){if(b==undefined)b=qj.length;qj[b]=a;return b|0}function
auf(a,b,c){var
d={};while(b){switch(b[1]){case
0:d.rdonly=1;break;case
1:d.wronly=1;break;case
2:d.append=1;break;case
3:d.create=1;break;case
4:d.truncate=1;break;case
5:d.excl=1;break;case
6:d.binary=1;break;case
7:d.text=1;break;case
8:d.nonblock=1;break}b=b[2]}if(d.rdonly&&d.wronly)Y(hw(a)+zK);if(d.text&&d.binary)Y(hw(a)+sz);var
e=aud(a),f=e.device.open(e.rest,d);return qk(f,undefined)}(function(){function
a(a,b){return lf()?at9(a,b):new
k_(a,b)}qk(a(0,{rdonly:1,altname:"/dev/stdin",isCharacterDevice:true}),0);qk(a(1,{buffered:2,wronly:1,isCharacterDevice:true}),1);qk(a(2,{buffered:2,wronly:1,isCharacterDevice:true}),2)}());function
atX(a){var
b=qj[a];if(b.flags.wronly)Y(Jd+a+" is writeonly");var
d=null,c={file:b,offset:b.flags.append?b.length():0,fd:a,opened:true,out:false,buffer_curr:0,buffer_max:0,buffer:new
Uint8Array(kB),refill:d};hx[c.fd]=c;return c.fd}function
L7(a){var
b=qj[a];if(b.flags.rdonly)Y(Jd+a+" is readonly");var
d=b.flags.buffered!==undefined?b.flags.buffered:1,c={file:b,offset:b.flags.append?b.length():0,fd:a,opened:true,out:true,buffer_curr:0,buffer:new
Uint8Array(kB),buffered:d};hx[c.fd]=c;return c.fd}function
atY(){var
b=0;for(var
a=0;a<hx.length;a++)if(hx[a]&&hx[a].opened&&hx[a].out)b=[0,hx[a].fd,b];return b}function
bh(a){a.t&6&&p_(a);return cw(a.c)}function
atZ(a,b,c,d){var
e=hx[a];if(!e.opened)Y("Cannot output to a closed channel");var
b=Mg(b);b=b.subarray(c,c+d);if(e.buffer_curr+b.length>e.buffer.length){var
g=new
Uint8Array(e.buffer_curr+b.length);g.set(e.buffer);e.buffer=g}switch(e.buffered){case
0:e.buffer.set(b,e.buffer_curr);e.buffer_curr+=b.length;g5(a);break;case
1:e.buffer.set(b,e.buffer_curr);e.buffer_curr+=b.length;if(e.buffer_curr>=e.buffer.length)g5(a);break;case
2:var
f=b.lastIndexOf(10);if(f<0){e.buffer.set(b,e.buffer_curr);e.buffer_curr+=b.length;if(e.buffer_curr>=e.buffer.length)g5(a)}else{e.buffer.set(b.subarray(0,f+1),e.buffer_curr);e.buffer_curr+=f+1;g5(a);e.buffer.set(b.subarray(f+1),e.buffer_curr);e.buffer_curr+=b.length-f-1}break}return 0}function
D1(a,b,c,d){return atZ(a,g4(b),c,d)}function
L8(a,b){var
c=cw(String.fromCharCode(b));D1(a,c,0,1);return 0}function
le(a,b){return+(p8(a,b,false)!=0)}function
D2(a,b){var
d=new
Array(b+1);d[0]=a;for(var
c=1;c<=b;c++)d[c]=0;return d}function
at1(a,b){a[0]=im;a[1]=b;return 0}function
iF(a){if(a
instanceof
Array&&a[0]==a[0]>>>0)return a[0];else if(DZ(a))return jr;else if(D0(a))return jr;else if(a
instanceof
Function||typeof
a=="function")return nZ;else if(a&&a.caml_custom)return kT;else
return kZ}function
ats(a){var
b;while(a)if(H(a[1][1])=="SYJS"){b=a[1][2];break}else
a=a[2];var
d={};if(b)for(var
c=1;c<b.length;c++)d[H(b[c][1])]=b[c][2];return d}function
b_(a,b,c){if(c){var
d=c;if(ak.toplevelReloc)a=p7(ak.toplevelReloc,[d]);else if(bg.toc){if(!bg.symbols)bg.symbols=ats(bg.toc);var
e=bg.symbols[d];if(e>=0)a=e;else
bf("caml_register_global: cannot locate "+d)}}bg[a+1]=b;if(c)bg[c]=b}function
D6(a,b){L_[hw(a)]=b;return 0}function
at4(a){a[2]=L$++;return a}function
V(a,b){if(a===b)return 1;return 0}function
at6(){aM(zk)}function
az(a,b){if(b>>>0>=J(a))at6();return u(a,b)}function
l(a,b){return 1-V(a,b)}function
at7(){return 0x7FFFFFFF/4|0}var
auc=ak.process&&ak.process.platform&&ak.process.platform==Ls?If:"Unix";function
at8(){return[0,cw(auc),32,0]}function
at2(){Mb(bg.Not_found)}function
Me(a){var
b=Mi(H(a));if(b===undefined)at2();return b$(b)}function
D7(a){var
b=1;while(a&&a.joo_tramp){a=a.joo_tramp.apply(null,a.joo_args);b++}return a}function
a1(a,b){return{joo_tramp:a,joo_args:b}}function
a(a,b){if(typeof
b==="function"){a.fun=b;return 0}if(b.fun){a.fun=b.fun;return 0}var
c=b.length;while(c--)a[c]=b[c];return 0}function
aA(a){{if(a
instanceof
Array)return a;var
b;if(ak.RangeError&&a
instanceof
ak.RangeError&&a.message&&a.message.match(/maximum call stack/i))b=bg.Stack_overflow;else if(ak.InternalError&&a
instanceof
ak.InternalError&&a.message&&a.message.match(/too much recursion/i))b=bg.Stack_overflow;else if(a
instanceof
ak.Error&&hZ(zr))b=[0,hZ(zr),a];else
b=[0,bg.Failure,b$(String(a))];if(a
instanceof
ak.Error)b.js_error=a;return b}}function
atP(a){switch(a[2]){case-8:case-11:case-12:return 1;default:return 0}}function
atz(a){var
b=w;if(a[0]==0){b+=a[1][1];if(a.length==3&&a[2][0]==0&&atP(a[1]))var
f=a[2],g=1;else
var
g=2,f=a;b+=KX;for(var
e=g;e<f.length;e++){if(e>g)b+=IL;var
d=f[e];if(typeof
d=="number")b+=d.toString();else if(d
instanceof
hr)b+=k7+d.toString()+k7;else if(typeof
d=="string")b+=k7+d.toString()+k7;else
b+=jf}b+=t9}else if(a[0]==c)b+=a[1];return b}function
L2(a){if(a
instanceof
Array&&(a[0]==0||a[0]==c)){var
d=hZ(KP);if(d)p7(d,[a,false]);else{var
e=atz(a),b=hZ(KH);if(b)p7(b,[0]);console.error(vx+e+"\n");if(a.js_error)throw a.js_error}}else
throw a}function
at5(){var
c=ak.process;if(c&&c.on)c.on("uncaughtException",function(a,b){L2(a);c.exit(2)});else if(ak.addEventListener)ak.addEventListener("error",function(a){if(a.error)L2(a.error)})}at5();function
k(a,b){return(a.l>=0?a.l:a.l=a.length)==1?a(b):ht(a,[b])}function
j(a,b,c){return(a.l>=0?a.l:a.l=a.length)==2?a(b,c):ht(a,[b,c])}function
t(a,b,c,d){return(a.l>=0?a.l:a.l=a.length)==3?a(b,c,d):ht(a,[b,c,d])}function
bu(a,b,c,d,e){return(a.l>=0?a.l:a.l=a.length)==4?a(b,c,d,e):ht(a,[b,c,d,e])}function
bF(a,b,c,d,e,f){return(a.l>=0?a.l:a.l=a.length)==5?a(b,c,d,e,f):ht(a,[b,c,d,e,f])}function
atm(a,b,c,d,e,f,g,h
